doc/api/fs.md

```diff
@@ -3546,6 +3544,12 @@ to the completion callback.
 Using `fs.rmdir()` on a file (not a directory) results in an `ENOENT` error on
 Windows and an `ENOTDIR` error on POSIX.
 
+Setting `recursive` to `true` results in behavior similar to the Unix command
+`rm -rf`: an error will not be raised for paths that do not exist, and paths
+that represent files will be deleted. The permissive behavior of the
+`recursive` option is deprecated, `ENOTDIR` and `ENOENT` will be thrown in
```

Comment:
@nodejs/tsc I thought it might be most elegant in `v16` that we make `rmdir`/`recursive` either match the behavior of .NET, Python, or Deno, with folks using `rm` if  the want looser (_rather than introducing yet another method_). Which is how I landed on this language...

I figure we have some time to get this perfect, with it being most important for `Node@v15` that we:

* try to land a new method `rm`.
* get the deprecation in place for folks using the existing approach.